lib/CGameState.cpp

```diff
@@ -3349,68 +3346,73 @@ void CPathfinder::calculatePaths()
 			turn++;
 		}
 
-
 		//add accessible neighbouring nodes to the queue
 		neighbours.clear();
 
-		//handling subterranean gate => it's exit is the only neighbour
-        bool subterraneanEntry = (ct->topVisitableId() == Obj::SUBTERRANEAN_GATE && useSubterraneanGates);
-		if(subterraneanEntry)
+		auto sObj = ct->topVisitableObj(cp->coord == CGHeroInstance::convertPosition(hero->pos, false));
+		auto cObj = dynamic_cast<const CGTeleport *>(sObj);
```

Comment:
Here should be check for successfull cast or at least comment about that all methods above can accept nullptr.